pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -23,7 +23,9 @@ class TypeHintMatch:
     return_type: list[str] | str | None | object
     # arg_types is for positional arguments
     arg_types: dict[int, str] | None = None
-    # kwarg_types is for the special case `**kwargs`
+    # named_arg_types is for named or keyword arguments
+    named_arg_types: dict[str, str] | None = None
+    # kwargs_type is for the special case `**kwargs`
     kwargs_type: str | None = None
```

Comment:
Just thinking, it could make sense to use actual variable docstrings here. I.e. below the variable definition.
```py
    arg_types: dict[int, str] | None = None
    """arg_types is for positional arguments"""
```

Reply:
Good idea! Done.